app/security/WebAuthnCredentialRepository.scala

```diff
@@ -0,0 +1,122 @@
+package security
+
+import models.user.{MultiUserDAO, WebAuthnCredential, WebAuthnCredentialDAO}
+import com.yubico.webauthn._
+import com.yubico.webauthn.data._
+import com.scalableminds.util.accesscontext.GlobalAccessContext
+import com.scalableminds.util.objectid.ObjectId
+import com.typesafe.scalalogging.LazyLogging
+import net.liftweb.common.{Empty, Failure, Full}
+
+import java.util.Optional
+import javax.inject.Inject
+import scala.annotation.nowarn
+import scala.jdk.CollectionConverters._
+
+object WebAuthnCredentialRepository {
+  def byteArrayToBytes(arr: ByteArray): Array[Byte] = arr.getBytes
+  def bytesToByteArray(bytes: Array[Byte]): ByteArray = new ByteArray(bytes)
+
+  def objectIdToByteArray(id: ObjectId): ByteArray = new ByteArray(id.toString.getBytes())
+  def byteArrayToObjectId(arr: ByteArray): ObjectId = new ObjectId(new String(arr.getBytes))
+}
+
+/*
+ * UserHandle => ObjectId
+ * Username => User's E-Mail address
+ */
+
+@nowarn("cat=deprecation")
+class WebAuthnCredentialRepository @Inject()(multiUserDAO: MultiUserDAO, webAuthnCredentialDAO: WebAuthnCredentialDAO)
+    extends CredentialRepository
+    with LazyLogging {
+  def getCredentialIdsForUsername(email: String): java.util.Set[PublicKeyCredentialDescriptor] = {
+    val result = for {
+      user <- multiUserDAO.findOneByEmail(email)(GlobalAccessContext).await("Java interop")
+      keys <- webAuthnCredentialDAO.findAllForUser(user._id)(GlobalAccessContext).await("Java interop")
+      creds = keys.map(key => {
+        PublicKeyCredentialDescriptor.builder().id(WebAuthnCredentialRepository.bytesToByteArray(key.keyId)).build()
+      })
+    } yield creds;
+    result match {
+      case Full(creds) => creds.toSet.asJava;
+      case Empty       => Set[PublicKeyCredentialDescriptor]().asJava;
+      case Failure(msg, _, _) => {
+        logger.error(msg);
+        Set[PublicKeyCredentialDescriptor]().asJava
+      }
+    }
+  }
+
+  def getUserHandleForUsername(email: String): Optional[ByteArray] =
+    multiUserDAO.findOneByEmail(email)(GlobalAccessContext).await("Java interop") match {
+      case Full(user) => Optional.ofNullable(WebAuthnCredentialRepository.objectIdToByteArray(user._id))
+      case Empty      => Optional.empty()
+      case Failure(msg, _, _) => {
+        logger.error(msg);
+        Optional.empty();
+      }
+    }
+
+  def getUsernameForUserHandle(handle: ByteArray): Optional[String] = {
```

Comment:
Can you please maybe add a comment that the handle is the encoded multiuser id? and needs to be named this way due to extending CredentialRepository?